src/Core/src/Hosting/MauiAppBuilder.cs

```diff
@@ -150,7 +150,9 @@ public MauiApp Build()
 				? _createServiceProvider()
 				: _services.BuildServiceProvider();
 
-			MauiApp builtApplication = new MauiApp(serviceProvider);
+			var appScope = serviceProvider.CreateScope();
```

Comment:
Why are we always creating a scope for the application? That doesn't seem necessary.

cc @davidfowl @halter73

Reply:
I feel like the core problem is really that some MAUI services have invalid scope dependencies (like a scoped service depending on a transient service or similar). And we never found that because we never had scope validation enabled. That's what my initial investigation from a long time ago revealed: https://github.com/dotnet/maui/issues/11457#issuecomment-1374076757

Reply:
This comes up for services that have the possibility to be scoped at the window or the application level. 

Currently we only create scopes for each window.

The scenario that @Eilon articulated in the original issue is probably the only current case where this is relevant (not sure if more will come down the road)

The `IDispatcher` is a scoped service that returns one dispatcher at the Application level and then a different dispatcher at the window level, depending on where you are and what you are trying to do. 

Before we've created any windows, the dispatcher at the application level is used. This is the exception that you see in @Eilon 's comment, because we're using the application-level dispatcher to perform operations. This dispatcher is also relevant for scenarios where you might not be inside a window, so they will fall back to the application service provider to retrieve the relevant `IDispatcher`

So, this case is relevant to any service that is registered as scoped but needs to resolve differently if it's from inside a window vs the application.  

The other approach could be to setup a scoped container for the app that's adjacent to the root container that we fall back to (instead of the root container) so that the app level container remains the root container and not scoped.

Reply:
Pushed up a variation of this PR based on the notes I made above ^

Reply:
I'm not sure if there are any negative implications to this so I"ll need to defer to @eerhardt and @davidfowl regarding always creating a scope.

Reply:
So even though `MauiApp.Services` remains an unscoped provider, the root `MauiContext.Services` now always has an application-level scope? What about code [already calling the public `MauiContext` ctor](https://github.com/AvaloniaUI/AvaloniaMauiHybrid/blob/7eb990fb60ccf96c9b14ad0856b3090d681eeb91/Avalonia.Maui/AvaloniaAppBuilderExtensions.cs#L72) with `MauiApp.Services`? Also, did we miss a call to the MauiContext ctor in [our own Tizen code](https://github.com/dotnet/maui/blob/babeeb28a828721cca9dc4bc16b5997c44e63138/src/Core/src/Platform/Tizen/MauiApplication.cs#L43)?

If this is the direction we want to go, I think we should update the API to be clearer about where it's exposing an application-level scoped provider vs unscoped root provider. Or at least update the doc comments and make a breaking change announcement.